docs/contributors/roadmap.md

```diff
@@ -1,48 +1,52 @@
-    * Persistent disks.
- 4. Keeping up with new Kubernetes releases and features.
- 5. Development-focused features like:
-   * Mounting host directories.
-   * VPN/proxy networking.
- 6. Native hypervisor integration.
- 7. Support for alternative Kubernetes runtimes, like rkt.
- 8. Removing the VirtualBox dependency and replacing it with Hypervisor.framework/Hyper-V.
-
-## Timelines
-Minikube will release much faster than this, so this section is fairly speculative.
-This section is subject to change based on feedback and staffing.
-
-### Q1 2017
-
-* Release Kubernetes 1.6.0 alpha and beta releases packaged with minikube
-* Release Kubernetes 1.6.0 packaged with minikube within two days of GA upstream build
-* Run local e2e Kubernetes tests with minikube
-* Minikube no longer depends on libmachine
-* Minikube no longer depends on existing KVM driver
-* Native drivers are made default and packaged with minikube
-* Improve minikube start time by 30%
-* Add a no-vm driver for linux CI environments
+# minikube roadmap (2019)
+
+This roadmap is a living document outlining the major technical improvements which we would like to see in minikube during 2019, divided by how they apply to the minikube [(guiding principles)[principles.md]
+
+Please send a PR to suggest any improvements to it.
+
+## (#1) User-friendly and accessible
+
+- Creation of a user-centric minikube website for installation & documentation
+- Localized output to 5+ written languages
+- Make minikube usable in environments with challenging connectivity requirements
+- Support lightweight deployment methods for environments where VM's are impractical
+- Add offline support
+
+## (#2) Inclusive and community-driven
+
+- Increase community involvement in planning and decision making
+- Make the continuous integration and release infrastructure publicly available
+- Double the number of active maintainers
+
+## (#3) Cross-platform
+
+- Simplified installation process across all supported platforms
+- Users should never need to separately install supporting binaries
+
+## (#4) Support all Kubernetes features
+
+- Add multi-node support
+
+## (#5) High-fidelity
+
+- Reduce guest VM overhead by 50%
+- Disable swap in the guest VM
+
+## (#6) Compatible with all supported Kubernetes releases
+
+- Continuous Integration testing across all supported Kubernetes releases
+- Automatic PR generation for updating the default Kubernetes release minikube uses
+
+## (#7) Support for all Kubernetes-friendly container runtimes
+
+- Run all integration tests across all supported container runtimes
+- Support for Kata Containers (help wanted!)
+
+## (#8) Stable and easy to debug
+
+- Pre-flight error checks for common connectivity and configuration errors
+- Improve the `minikube status` command so that it can diagnose common issues
+- Mark all features not covered by continuous integration as `experimental`
+- Replace built-in machine drivers (virtualbox, kvm2) with their upstream equivalents
```

Comment:
AFAIK, the "virtualbox" driver is already built-in (to machine) and "kvm2" doesn't have an upstream (from docker).
Replacing kvm2 with kvm sounds like a step backwards, better to continue to evolve the minikube KVM driver ?

Reply:
Thanks. This was supposed to say hyperkit and kvm2. I'm just going to remove this from the technical roadmap, since it's strictly about technical debt reduction and is not terribly inspirational or important to end users.

The idea is to push our driver improvements upstream so that we may share improvements with minishift and other consumers:

kvm -> https://github.com/machine-drivers/docker-machine-driver-libvirt
hyperkit -> https://github.com/machine-drivers/docker-machine-driver-hyperkit

But only if doing so moves minikube forwards rather than backwards.

Thanks for the input!